From 3f40894a8b6074749d670924c9542ef5c6ec65ec Mon Sep 17 00:00:00 2001
From: Jari Nippula <jari.nippula@intel.com>
Date: Thu, 12 Jan 2017 14:03:32 +0200
Subject: [PATCH] Revert patch: "[REVERTME] mmc: host: pci: disable
 functionality to help enumeration" Fix emmc write fails in some bxt devices
 supporting only DDR50 emmc speed.

Signed-off-by: Jari Nippula <jari.nippula@intel.com>
---
 drivers/mmc/host/sdhci-pci-core.c | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/mmc/host/sdhci-pci-core.c b/drivers/mmc/host/sdhci-pci-core.c
index 83ec752..3a0b6b5 100644
--- a/drivers/mmc/host/sdhci-pci-core.c
+++ b/drivers/mmc/host/sdhci-pci-core.c
@@ -357,18 +357,6 @@ out:
 	return ret;
 }
 
-
-static int bxt_emmc_probe_slot(struct sdhci_pci_slot *slot)
-{
-	slot->host->mmc->caps |= MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE |
-				 MMC_CAP_1_8V_DDR | MMC_CAP_WAIT_WHILE_BUSY;
-
-	slot->host->mmc->caps2 |= MMC_CAP2_HC_ERASE_SZ | MMC_CAP2_HS400_1_8V |
-				  MMC_CAP2_HS200_1_8V_SDR;
-
-	return 0;
-}
-
 static int byt_emmc_probe_slot(struct sdhci_pci_slot *slot)
 {
 	slot->host->mmc->caps |= MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE |
@@ -408,13 +396,6 @@ static int byt_sd_probe_slot(struct sdhci_pci_slot *slot)
 	return 0;
 }
 
-static const struct sdhci_pci_fixes sdhci_intel_bxt_emmc = {
-	.allow_runtime_pm = true,
-	.quirks2	= SDHCI_QUIRK2_PRESET_VALUE_BROKEN |
-			  SDHCI_QUIRK2_CAPS_BIT63_FOR_HS400,
-	.probe_slot     = bxt_emmc_probe_slot,
-};
-
 static const struct sdhci_pci_fixes sdhci_intel_byt_emmc = {
 	.allow_runtime_pm = true,
 	.probe_slot	= byt_emmc_probe_slot,
@@ -1195,7 +1176,7 @@ static const struct pci_device_id pci_ids[] = {
 		.device		= PCI_DEVICE_ID_INTEL_BXTM_EMMC,
 		.subvendor	= PCI_ANY_ID,
 		.subdevice	= PCI_ANY_ID,
-		.driver_data	= (kernel_ulong_t)&sdhci_intel_bxt_emmc,
+		.driver_data	= (kernel_ulong_t)&sdhci_intel_byt_emmc,
 	},
 
 	{
-- 
2.7.4

